### DISCUSSION TITLE
dns queries always goes through tunnel

### DISCUSSION THREAD

**Discussion by naox** [wiresock/WireSockUI#58] [General]
DNS queries (from every program, whole OS) seems to always  always goes through tunnel, even if you just specify one program in AllowedApps (when DNS is specified in tunnel config). Specifying dns IPs in DisallowedIPs does not change anything. Specifying DisallowedApps = svhost does not change anything. This might not be thing one wants because dns servers like cloudflare, google, opendns use geolocation and thus every dns request from OS goes to dns in geolocation of tunnel Endpoint, which might be far away and whole OS starts lagging because of this. And what is more important when Endpoint goes down dns resolution in whole OS goes down.

**Comment by wiresock** [comment]
DNS queries in Windows are invariably processed through the DNSCACHE process, which means they cannot be distinguished by the originating applications. To address this, you can either eliminate the DNS settings from your Wireguard configuration or include DNSCACHE in the DisallowedApps list.

**Reply by naox** [reply]
Thanks.

Not setting DNS in tunnel config causes dns resolution for app in AllowedApps to not work.

Setting 
`DisallowedApps = DNSCACHE`
works


You might consider puting that info info wiresock documentation as its basicly critical for using AllowedApps on Windows
